pkg/theia/commands/anomaly_detection_run_test.go

```diff
@@ -107,3 +148,215 @@ func TestAnomalyDetectionRun(t *testing.T) {
+			cmd.Flags().Int32("executor-instances", 1, "")
+			cmd.Flags().String("driver-core-request", "1", "")
+			cmd.Flags().String("driver-memory", "mock_driver-memory", "")
+		case "Unspecified executor-core-request":
+			cmd.Flags().String("algo", "ARIMA", "")
+			cmd.Flags().String("start-time", "2006-01-02 15:04:05", "")
+			cmd.Flags().String("end-time", "2006-01-03 16:04:05", "")
+			cmd.Flags().String("ns-ignore-list", "[\"kube-system\",\"flow-aggregator\",\"flow-visibility\"]", "")
+			cmd.Flags().Int32("executor-instances", 1, "")
+			cmd.Flags().String("driver-core-request", "1", "")
+			cmd.Flags().String("driver-memory", "1m", "")
+		case "Invalid executor-core-request":
+			cmd.Flags().String("algo", "ARIMA", "")
+			cmd.Flags().String("start-time", "2006-01-02 15:04:05", "")
+			cmd.Flags().String("end-time", "2006-01-03 16:04:05", "")
+			cmd.Flags().String("ns-ignore-list", "[\"kube-system\",\"flow-aggregator\",\"flow-visibility\"]", "")
+			cmd.Flags().Int32("executor-instances", 1, "")
+			cmd.Flags().String("driver-core-request", "1", "")
+			cmd.Flags().String("driver-memory", "1m", "")
+			cmd.Flags().String("executor-core-request", "mock_executor-core-request", "")
+		case "Unspecified executor-memory":
+			cmd.Flags().String("algo", "ARIMA", "")
+			cmd.Flags().String("start-time", "2006-01-02 15:04:05", "")
+			cmd.Flags().String("end-time", "2006-01-03 16:04:05", "")
+			cmd.Flags().String("ns-ignore-list", "[\"kube-system\",\"flow-aggregator\",\"flow-visibility\"]", "")
+			cmd.Flags().Int32("executor-instances", 1, "")
+			cmd.Flags().String("driver-core-request", "1", "")
+			cmd.Flags().String("driver-memory", "1m", "")
+			cmd.Flags().String("executor-core-request", "1", "")
+		case "Invalid executor-memory":
+			cmd.Flags().String("algo", "ARIMA", "")
+			cmd.Flags().String("start-time", "2006-01-02 15:04:05", "")
+			cmd.Flags().String("end-time", "2006-01-03 16:04:05", "")
+			cmd.Flags().String("ns-ignore-list", "[\"kube-system\",\"flow-aggregator\",\"flow-visibility\"]", "")
+			cmd.Flags().Int32("executor-instances", 1, "")
+			cmd.Flags().String("driver-core-request", "1", "")
+			cmd.Flags().String("driver-memory", "1m", "")
+			cmd.Flags().String("executor-core-request", "1", "")
+			cmd.Flags().String("executor-memory", "mock_executor-memory", "")
+		case "Unspecified use-cluster-ip":
+			cmd.Flags().String("algo", "ARIMA", "")
+			cmd.Flags().String("start-time", "2006-01-02 15:04:05", "")
+			cmd.Flags().String("end-time", "2006-01-03 16:04:05", "")
+			cmd.Flags().String("ns-ignore-list", "[\"kube-system\",\"flow-aggregator\",\"flow-visibility\"]", "")
+			cmd.Flags().Int32("executor-instances", 1, "")
+			cmd.Flags().String("driver-core-request", "1", "")
+			cmd.Flags().String("driver-memory", "1m", "")
+			cmd.Flags().String("executor-core-request", "1", "")
+			cmd.Flags().String("executor-memory", "1m", "")
```

Comment:
Hi Tushar, I noticed that there are many duplicate flag settings code between lines 241-352. To improve this, one possible approach that comes to mind is to preset a cmd variable with all correct flags, and then in each test case, use `markHidden` to remove/hidden a flag for the "Unspecified xxxx" test cases and set the flag to a mock value for the "Invalid xxxx" test cases. This would make the code look cleaner and more organized in my opinion.

Reply:
currently we can not find an approach to reset the flags, resolving this conversation for now